docs/reference/search/search-your-data/knn-search.asciidoc

```diff
@@ -20,9 +20,11 @@ Common use cases for kNN include:
 === Prerequisites
 
 * To run a kNN search, you must be able to convert your data into meaningful
-vector values. You create these vectors outside of {es} and add them to
-documents as <<dense-vector,`dense_vector`>> field values. Queries are
-represented as vectors with the same dimension.
+vector values. You can
+{ml-docs}/ml-nlp-text-emb-vector-search-example.html[create these vectors using
+a natural language processing (NLP) model in {es}], or generate them outside
+{es}, and add them to documents as <<dense-vector,`dense_vector`>> field values.
```

Comment:
Nit: This is a pretty complex sentence. I'd consider breaking it after "generate them outside {es}"

Reply:
Done. Thank James!